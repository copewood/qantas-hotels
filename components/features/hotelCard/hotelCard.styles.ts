import styled from 'styled-components';

export const StyledCardContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  width: 100%;
  padding-bottom: 2rem;
  border-bottom: 1px solid black;
  margin-bottom: 1rem;
`;

export const StyledHotelInfo = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  width: 100%
`;

export const StyledPricingInfo = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  width: 100%
`;

export const StyledPropertyName = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 20px;
`;

export const StyledPropertyAddress = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 14px;
  & span {
    margin-left: 4px;
  }
`;

export const StyledOffer = styled.a`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 20px;
  color: red;
  text-decoration: underline;
`;

export const StyledCancellation = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 18px;
  color: green;
`;


export const StyledPrice = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 26px;
`;

export const StyledSaving = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
  font-size: 18px;
  color: red;
`;

