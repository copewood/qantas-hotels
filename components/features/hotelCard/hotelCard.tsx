import React, { FC } from 'react';

import { HotelImage } from '../../../components/atoms/hotelImage/hotelImage';
import { StarRating } from '../../../components/atoms/starRating/starRating';

import { 
  StyledCardContainer,
  StyledHotelInfo,
  StyledPricingInfo,
  StyledPropertyName,
  StyledPropertyAddress,
  StyledOffer,
  StyledCancellation,
  StyledPrice,
  StyledSaving
} from './hotelCard.styles';
import { CANCELLATION_OPTION, Hotel } from '../../models/hotel';

type HotelCardProps = {
  hotel: Hotel;
};

export const HotelCard: FC<HotelCardProps> = ({hotel}) => {
  const { property, offer } = hotel;
  return <>
    <StyledCardContainer>
      <HotelImage previewImage={property.previewImage} offer={offer}/>
      <StyledHotelInfo>
        <StyledPropertyName>
          <span>{property.title}</span>
          <StarRating rating={property.rating}/>
        </StyledPropertyName>
        <StyledPropertyAddress>
          {property.address.map((address: string, index: number) => 
            <span key={`${hotel.id}-${index}`}>{address}</span>)
          }
        </StyledPropertyAddress>
        <StyledOffer href="/">
          {offer.name}
        </StyledOffer>
        <StyledCancellation>
          {offer.cancellationOption.cancellationType === CANCELLATION_OPTION.FREE_CANCELLATION && 
            <span>Free Cancellation</span>}
        </StyledCancellation>     
      </StyledHotelInfo>
      <StyledPricingInfo>
        <div>1 night total (AUD)</div>
        <StyledPrice>
          ${offer.displayPrice.amount}
        </StyledPrice>
        {offer.savings && offer.savings.amount && 
          <StyledSaving>
            Save ${offer.savings.amount}
          </StyledSaving>}
      </StyledPricingInfo>
    </StyledCardContainer>
  </>;
};
