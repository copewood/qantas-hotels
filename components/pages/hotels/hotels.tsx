import React, { FC, useEffect, useState} from 'react';

import { HotelListHeader } from 'components/features/hotelListHeader/hotelListHeader';
import { HotelList } from 'components/features/hotelList/hotelList';
import { Hotel } from '@components/models/hotel';

type HotelsProps = {};

const fetchHotels = async () => {
    return fetch('/api/hotels')
        .then((response) => response.json())
        .then((responseJson) => responseJson);
}

export const Hotels: FC<HotelsProps> = ({}) => {
  const [hotels, setHotels] = useState<Array<Hotel>>();

  useEffect(() => {
    fetchHotels().then((response)=>{
      setHotels(response.results as Array<Hotel>)
    });
  },[])
  
  return <>
    {hotels && <>
      <HotelListHeader hotels={hotels}/>
      {/* <HotelList hotels={hotels}/> */}
    </>}
  </>;
};
