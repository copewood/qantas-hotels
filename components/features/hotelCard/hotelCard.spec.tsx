import React from 'react';
import { render } from '@testing-library/react';
import { HotelCard } from './hotelCard';
import { mockHotels } from '../../../pages/api/hotels'

describe('HotelList', () => {
  it('should show HotelCard', () => {
    const { container } = render(<HotelCard hotel={mockHotels.results[0]}/>);
    expect(container).toMatchSnapshot()
  });
});
