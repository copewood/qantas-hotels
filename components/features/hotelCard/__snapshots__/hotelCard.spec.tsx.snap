// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList should show HotelCard 1`] = `
<div>
  <div
    class="sc-eCYdqJ tFDdR"
  >
    <div
      class="sc-bczRLJ bxsSWL"
    >
      <img
        class="sc-gsnTZi bFjRRX"
        src="https://unsplash.it/145/125/?random"
      />
      <div
        class="sc-dkzDqf iuODgQ"
      >
        Exclusive Deal
      </div>
    </div>
    <div
      class="sc-jSMfEi jAQJee"
    >
      <div
        class="sc-iBkjds dqsmth"
      >
        <span>
          Courtyard by Marriott Sydney-North Ryde
        </span>
        <span
          class="sc-hKMtZM gBiuEH"
        >
          Star rating 4.5
        </span>
      </div>
      <div
        class="sc-ftvSup hGwNMW"
      >
        <span>
          7-11 Talavera Rd
        </span>
        <span>
          North Ryde
        </span>
      </div>
      <a
        class="sc-papXJ hierex"
        href="/"
      >
        Deluxe Balcony Room
      </a>
      <div
        class="sc-jqUVSM cERLEw"
      />
    </div>
    <div
      class="sc-gKXOVf eDpLKX"
    >
      <div>
        1 night total (AUD)
      </div>
      <div
        class="sc-kDDrLX DgOrb"
      >
        $
        329
      </div>
      <div
        class="sc-iqcoie esPlUa"
      >
        Save $
        30
      </div>
    </div>
  </div>
</div>
`;
